# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/UseCounters.conf
# Commit: 1dd1545eb1f9
# Full Hash: 1dd1545eb1f90ae9a8845c735ca6f173fafbf5a4
# Author: Matthew Gaudet <mgaudet@mozilla.com>
# Date: 2025-05-27 16:03:10
# Regressor Bug: 1964451
# File Overlap Count: 1
# Description:
#   Bug 1964451 - Add use counters for OOM handling r=sfink,mccr8
#   
#   So the goal here is that we get OOM data where processes survive long enough
#   send telemetry. To get the case where processes -crash- we'll also have to
#   get crash data. See Bug 1964092 for that half.
# ==============================================================================

diff -r 1d7f4558293e -r 1dd1545eb1f9 dom/base/UseCounters.conf
--- a/dom/base/UseCounters.conf	Mon May 26 17:09:42 2025 +0000
+++ b/dom/base/UseCounters.conf	Mon May 26 17:41:56 2025 +0000
@@ -85,6 +85,10 @@
 custom JS_dateparse called Date.parse
 custom JS_dateparse_impl_def call to Date.parse used implementation defined behaviour
 custom JS_regexp_symbol_protocol_on_primitive passed a primitive with a custom implementation of the regexp protocol symbols to one of the String methods that accept a regexp
+custom JS_large_oom_reported reported a large oom
+custom JS_small_oom_reported reported a small oom
+custom JS_large_oom_recovered recovered a large oom
+custom JS_small_oom_recovered recovered a small oom
 
 // Console API
 method console.assert